test/e2e/framework/providers/gce/gce.go

```diff
@@ -179,7 +179,7 @@ func (p *Provider) EnsureLoadBalancerResourcesDeleted(ip, portRange string) erro
 
 func getGCEZoneForGroup(group string) (string, error) {
 	zone := framework.TestContext.CloudConfig.Zone
-	if framework.TestContext.CloudConfig.MultiZone {
+	if framework.TestContext.CloudConfig.MultiZone || framework.TestContext.CloudConfig.Region != "" {
```

Comment:
IIUC this means that we're assuming we're creating a regional cluster if `CloudConfig.Region != ""`. Is that a safe assumption? (what if someone sets it to a non-empty value?)

To avoid above hassle, WDYT about always querying for the zone?

Reply:
Done.